client/config.go

```diff
@@ -64,12 +79,44 @@ func (c *Config) ValidateAndSetDefaults() error {
 	if c.Timeout < time.Millisecond {
 		c.Timeout = 10 * time.Second
 	}
+	if c.HasCustomDNSResolver() {
+		_, err := c.ParseDNSResolver()
+		if err != nil {
+			return ErrInvalidDNSResolver
+		}
+	}
 	if c.HasOAuth2Config() && !c.OAuth2Config.isValid() {
 		return ErrInvalidClientOAuth2Config
 	}
 	return nil
 }
 
+// Returns true if the DNSResolver is set in the configuration
+func (c *Config) HasCustomDNSResolver() bool {
+	return len(c.DNSResolver) > 0
+}
+
+// Parses the DNSResolver configuration string into the DNSResolverConfig struct
+func (c *Config) ParseDNSResolver() (DNSResolverConfig, error) {
+	re := regexp.MustCompile(`^(?P<proto>(.*))://(?P<host>[A-Za-z0-9\-\.]+):(?P<port>[0-9]+)?(.*)$`)
+	matches := re.FindStringSubmatch(c.DNSResolver)
+	if len(matches) == 0 {
+		return DNSResolverConfig{}, errors.New("ParseError")
+	}
+	r := make(map[string]string)
+	for i, k := range re.SubexpNames() {
+		if i != 0 && k != "" {
+			r[k] = matches[i]
+		}
+	}
+
+	return DNSResolverConfig{
+		Protocol: r["proto"],
+		Host:     r["host"],
+		Port:     r["port"],
+	}, nil
+}
```

Comment:
Is there really a need to so thoroughly parse the string?

Also, do we really need to pass the protocol here? I'd almost rather just default to TCP as a protocol all the time.

Reply:
Many (corporate) DNS servers only allow UDP and (some) cloud DNS services only accept TCP, so I think we should be able to choose accordingly.

Would you like it more if we had a nested configuration?

```yaml
dns-resolver:
  protocol: udp
  host: 127.0.0.1
  port: 53
```

Reply:
That looks a little overkill to me. Let's stay with what you currently made; after all, the whole `protocol://host:port` is used all over Gatus already. Worst case scenario, I'm planning on releasing v4.0.0 soon, so if I want to change something, I'll do it when that happens.